src/daft-sql/src/modules/aggs.rs

```diff
@@ -103,6 +103,7 @@ fn handle_count(inputs: &[FunctionArg], planner: &SQLPlanner) -> SQLPlannerResul
 pub fn to_expr(expr: &AggExpr, args: &[ExprRef]) -> SQLPlannerResult<ExprRef> {
     match expr {
         AggExpr::Count(_, _) => unreachable!("count should be handled by by this point"),
+        AggExpr::CountDistinct(..) => todo!("count distinct"),
```

Comment:
@universalmind303, I'll need a pointer on how to implement this part right here. Not sure if `CouldDistinct` should already be handled or not.

Reply:
we'll either have to special case `count distinct`, or propagate the distinct clause down through `to_expr`. The distinct clause is handled [here](https://github.com/universalmind303/Daft/blob/09cac7be10b34ce8eb922a90a6d74ba7b4dae619/src/daft-sql/src/functions.rs#L275)

Reply:
I'm not sure that I follow... What do you mean by "special-case" count-distinct? Do you mean that we change it to `unimplemented!(..)`?

Reply:
Not sure why/where `Count` is handled earlier on in the callstack. Might need to pair with you on this one?

Reply:
by special case i mean specifically handling `count(distinct ..)` at the call site [here](https://github.com/universalmind303/Daft/blob/09cac7be10b34ce8eb922a90a6d74ba7b4dae619/src/daft-sql/src/functions.rs#L275). 

> Not sure why/where `Count` is handled earlier on in the callstack. Might need to pair with you on this one? 

it's currently not, but handling it earlier would solve this problem without cluttering up the `to_expr` function. 


the alternative would be to propagate the distinct clause down to [handle_count](https://github.com/universalmind303/Daft/blob/09cac7be10b34ce8eb922a90a6d74ba7b4dae619/src/daft-sql/src/modules/aggs.rs#L72). That means that we'd have to pass the distinct clause through [SQLFunction::to_expr](https://github.com/universalmind303/Daft/blob/09cac7be10b34ce8eb922a90a6d74ba7b4dae619/src/daft-sql/src/functions.rs#L68) which I'm not a huge fan of passing that through for one case.

so I think both solutions have their own downsides. 

let me know if you have any other questions!

Reply:
Ah okay, I'll take a crack at it. Thanks!

Reply:
Talked with Jay offline offline about this. Don't want this blocking the PR. Will create a follow-up PR for this. Resolving this for now.